storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3851,22 +3851,6 @@ static dberr_t fsp_get_sys_used_segment(inode_info *inodes, mtr_t *mtr)
                               TRX_RSEG_FSEG_HEADER))
         err= DB_CORRUPTION;
       block->page.unfix();
-
-      /* Even after slow shutdown, there is a possiblity that
-      cached undo log can exist. So store the segment as used one */
-      for (trx_undo_t *undo= UT_LIST_GET_FIRST(rseg->undo_cached);
-           undo && err == DB_SUCCESS;
-           undo= UT_LIST_GET_NEXT(undo_list, undo))
-      {
-        block= buf_pool.page_fix(page_id_t{0, undo->hdr_page_no}, &err,
-                                 buf_pool_t::FIX_WAIT_READ);
-        if (!block)
-	  return err;
-	if (!inodes->insert_seg(block->page.frame + TRX_UNDO_SEG_HDR +
-                                TRX_UNDO_FSEG_HEADER))
-          err= DB_CORRUPTION;
```

Comment:
I checked that if this code is not removed, the test would fail like this:
```diff
CURRENT_TEST: innodb.undo_leak
--- /home/marko/11.2/mysql-test/suite/innodb/r/undo_leak.result	2024-11-15 06:37:39.933929352 -0800
+++ /home/marko/11.2/mysql-test/suite/innodb/r/undo_leak.reject	2024-11-15 06:52:50.204918531 -0800
@@ -43,4 +43,5 @@
 # restart
 SELECT NAME, FILE_SIZE FROM information_schema.innodb_sys_tablespaces WHERE SPACE = 0;
 NAME	FILE_SIZE
-innodb_system	12582912
+Warnings:
+Warning	1012	InnoDB: SELECTing from INFORMATION_SCHEMA.innodb_sys_tablespaces but the InnoDB storage engine is not installed
```